profiler/profiler.go

```diff
@@ -303,10 +304,20 @@ func (p *profiler) collect(ticker <-chan time.Time) {
 		p.seq++
 
 		completed = completed[:0]
+		for _, t := range p.enabledProfileTypes() {
+			if t != CPUProfile {
+				// pendingProfiles should be updated before
+				// starting the CPU profiler
+				p.pendingProfiles.Add(1)
+			}
+		}
```

Comment:
NIT: Any reason for duplicating the loop below and not just putting this code block below line 315?

Reply:
Yes, mainly because I'm paranoid and want to know it's correct 😅 . The loop below this is starting each profile type, including the CPU profile, in a new goroutine. I want the CPU profile to only stop when every other profile is stopped. I know this will be the case if the  CPU profile _starts_ with `pendingProfiles` incremented appropriately. Of course, realistically the loop that starts the profiles is going to be done before the CPU profile is done sleeping and hits `pendingProfiles.Wait()`. If this is excessively paranoid, I merge this loop into the loop below.

Reply:
Oh I see, subtle! And not paranoid, I think it's worth considering. Is that what the comment in loop is trying to explain? If yes, I think spelling it out in a bit more detail and perhaps attaching it to the loop (as it explains the reason why it exists) might give this a better chance to be "refactored" by a well-meaning person in the future with the same idea as me : p.

Alternatively you just do what I proposed, but add this before the loop:

```go
p.pendingProfiles.Add(1)
defer p.pendingProfiles.Done()
```

But I think the extra loop is less hacky and more clear.

Reply:
👍 Agreed, it is subtle. I updated the comment to explain why it's done the way it is.